packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -117,21 +138,62 @@ class _PopupMenu<T> extends StatelessComponent {
   }
 }
 
+class _PopupMenuRouteLayout extends OneChildLayoutDelegate {
+  _PopupMenuRouteLayout(this.position, this.selectedIndex);
+
+  final ModalPosition position;
+  final int selectedIndex;
+
+  BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+    return new BoxConstraints(
+      minWidth: 0.0,
+      maxWidth: constraints.maxWidth,
+      minHeight: 0.0,
+      maxHeight: constraints.maxHeight
+    );
+  }
+
+  // Put the child wherever position specifies, so long as it will fit within the
+  // specified parent size padded (inset) by 8. If necessary, adjust the child's
+  // position so that it fits.
+  Offset getPositionForChild(Size size, Size childSize) {
+    double x = position?.left ?? (position?.right != null ? size.width - (position.right + childSize.width) : 8.0);
+    double y = position?.top ?? (position?.bottom != null ? size.height - (position.bottom - childSize.height) : 8.0);
+
+    if (selectedIndex != -1)
+      y -= (_kMenuItemHeight * selectedIndex) + _kMenuVerticalPadding + _kMenuItemHeight / 2.0;
```

Comment:
you don't have to fix this in this patch, but if you don't, please either put a TODO or file a bug:

For things with a selectedIndex, if the menu is moved, we should ensure that the Y position of the children are maintained even if the container is moved up, resulting in the contents of the menu being over-scrolled.